packages/style-engine/class-wp-style-engine-css-declarations.php

```diff
@@ -17,6 +17,12 @@
  * @access private
  */
 class WP_Style_Engine_CSS_Declarations {
+	/**
+	 * An array of valid CSS custom properties.
+	 */
+	const VALID_CUSTOM_PROPERTIES = array(
+		'--wp--style--unstable-gallery-gap',
+	);
```

Comment:
Just an idea, but could this be a list of key/pairs of the custom property and the desired value to test against? E.g.

```
const VALID_CUSTOM_PROPERTIES = array(
	'--wp--style--unstable-gallery-gap' => 'gap',
);
```

Then in `filter_declaration` you could use the `! empty( safecss_filter_attr( ...` check from the previous commit, passing in the property from the `VALID_CUSTOM_PROPERTIES` lookup when doing the check?

Reply:
Hmmm this is tricky... The problem is that we don't really have a list of valid CSS props! Users are able to define their own custom CSS properties in theme.json (see settings.custom in https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/themes/theme-json.md#css-custom-properties-presets--custom), so while this may be a suitable solution now, it probably won't work long-term once we get to global-styles! It will also require a lot of maintenance as we move a lot of things to css-vars...

Reply:
> The problem is that we don't really have a list of valid CSS props! Users are able to define their own custom CSS properties in theme.json

Yeah, all good points 🤔 

It's part of the reason why I went for the lazy option in 33e336a9ae8c627f1e3afdc9575f97e8fbce0de1 😄 

I mean this PR as it will resolve https://github.com/WordPress/gutenberg/pull/43070, but I agree we'll have to come up with something a little more robust down the road.

Reply:
> Then in filter_declaration you could use the ! empty( safecss_filter_attr( ... check from the previous commit, passing in the property from the VALID_CUSTOM_PROPERTIES lookup when doing the check?

I've done this for now 👍 

If we need to support global styles before core, if ever, supports CSS custom properties maybe we can revert back to https://github.com/WordPress/gutenberg/commit/33e336a9ae8c627f1e3afdc9575f97e8fbce0de1